lib/Conf.ml

```diff
@@ -978,6 +981,22 @@ module Formatting = struct
         update conf ~f:(fun f -> {f with let_binding_spacing= elt}) )
       (fun conf -> conf.fmt_opts.let_binding_spacing)
 
+  let let_binding_punning =
+    let doc = "Name punning in extended let bindings" in
+    let names = ["let-binding-punning"] in
+    let all =
+      [ Decl.Value.make ~name:"preserve" `Preserve
+          "$(b,preserve) uses let-punning only when it exists in the source."
+      ; Decl.Value.make ~name:"always" `Always
+          "$(b,always) uses let-punning whenever possible."
+      ; Decl.Value.make ~name:"never" `Never
+          "$(b,never) never uses let-punning." ]
```

Comment:
To make the difference between never and preserve clearer, I think you should say that existing code using let-punning will be rewritten without.